controllers/slicegateway/reconciler.go

```diff
@@ -192,8 +192,75 @@ func (r *SliceGwReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 			)
 			return ctrl.Result{}, err
 		}
+		//reconcile the available nodes
+		if err := r.reconcileNodes(ctx, sliceGw); err != nil {
+			return ctrl.Result{}, err
+		}
 	}
 
+	isClient := sliceGw.Status.Config.SliceGatewayHostType == "Client"
+	if isClient {
+		//check if SliceGatewayRemoteGatewayID is populated , else requeue
+		if sliceGw.Status.Config.SliceGatewayRemoteNodeIP == "" || sliceGw.Status.Config.SliceGatewayRemoteGatewayID == "" {
+			log.Info("slicegateway waiting for SliceGatewayRemoteGatewayID to get populate from controller")
+			return ctrl.Result{
+				RequeueAfter: time.Duration(10) * time.Second,
+			}, nil
+		}
+		//create a headless service and an endpoint for DNS Query on OpenVPN Client
+		serviceFound := corev1.Service{}
+		err = r.Get(ctx, types.NamespacedName{Namespace: sliceGw.Namespace, Name: sliceGw.Status.Config.SliceGatewayRemoteGatewayID}, &serviceFound)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Create a new service with same name as SliceGatewayRemoteGatewayID , because --remote flag of openvpn client is populated with same name. So it would call this svc to get a server IP(through endpoint)
+				svc := r.createHeadlessServiceForGwServer(sliceGw)
+				log.Info("Creating a new Headless Service")
+				if err := r.Create(ctx, svc); err != nil {
+					log.Error(err, "Failed to create a new headless service", "Name", svc.Name)
+					return ctrl.Result{}, err
+				}
+				log.Info("Created a headless service")
+				return ctrl.Result{Requeue: true}, nil
+			} else {
+				log.Error(err, "Failed to Get Headless service")
+				return ctrl.Result{}, err
+			}
+		}
+		//create an endpoint if not exists
+		endpointFound := corev1.Endpoints{}
+		err := r.Get(ctx, types.NamespacedName{
+			Namespace: sliceGw.Namespace,
+			Name:      sliceGw.Status.Config.SliceGatewayRemoteGatewayID}, &endpointFound)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				//Create a new endpoint with same name as RemoteGatewayID
+				endpoint := r.createEndpointForGatewayServer(sliceGw)
+				log.Info("Creating a new Endpoint for Headless Service")
+				if err := r.Create(ctx, endpoint); err != nil {
+					log.Error(err, "Failed to create an Endpoint", "Name", endpoint.Name)
+					return ctrl.Result{}, err
+				}
+				log.Info("Created a new Endpoint for sliceGw", "sliceGw", sliceGw.Name)
+				return ctrl.Result{Requeue: true}, nil
+			} else {
+				log.Error(err, "Failed to get an endpoint")
+				return ctrl.Result{}, err
+			}
+		}
+		debugLog.Info("SliceGatewayRemoteNodeIP", "SliceGatewayRemoteNodeIP", sliceGw.Status.Config.SliceGatewayRemoteNodeIP)
+		debugLog.Info("from endpointFound", "from endpointFound", endpointFound.Subsets[0].Addresses[0].IP)
```

Comment:
This log can be moved inside the 'if' block on line 253. The log on line 254 can print the value in endpointFound.